arch/arm/src/tlsr82/tlsr82_flash_mtd.c

```diff
@@ -292,13 +292,16 @@ static void tlsr82_flash_print(const char *msg, const uint8_t *buf,
     {
       if (i % 16 == 0)
         {
-          off += sprintf(&print_buf[off], "0x%08x:", i);
+          snprintf(&print_buf[off],
+                   sizeof(print_buf) - off, "0x%08x:", i);
+          off += strlen(&print_buf[off]);
         }
 
-      off += sprintf(&print_buf[off], "0x%02x ", buf[i]);
-      i++;
+      snprintf(&print_buf[off],
+               sizeof(print_buf) - off, "0x%02x ", buf[i]);
+      off += strlen(&print_buf[off]);
 
-      if (i % 16 == 0)
+      if (++i % 16 == 0)
```

Comment:
Optional: As an alternative the `while (len--)` can be changed to
```
  fo (i = 0; len-- > 0; i++)
    {
    ...
    }
```

Reply:
but the condition need be changed:
```
if (i % 16 == 15)
```

Reply:
yes, indeed